### PR TITLE
lib/run_part: Minor cleanups

### PR DESCRIPTION
**Pull request by stoeckmann**
- Use correct data type
- Unify logging (at least the fprintf calls)
- Reduce visibility of locally used function

### CODE REVIEW

File: lib/run_part.c
Comment (alejandro-colomar) [inline lib/run_part.c:65]: But there are also perror(3) calls.  The commit should also justify why those still use stderr.

File: lib/run_part.c
Comment (alejandro-colomar) [inline lib/run_part.c:18]: LGTM.

File: lib/run_part.c
Comment (alejandro-colomar) [inline lib/run_part.c:22]: LGTM

File: lib/run_part.c
Comment (stoeckmann) [reply lib/run_part.c]: Or I just replace them as well. ;) Done.

Comment (alejandro-colomar) [reply lib/run_part.c]: I'd like at least a justification of why not unify the other way around.  :)

Comment (stoeckmann) [reply lib/run_part.c]: https://github.com/shadow-maint/shadow/blob/master/lib/shadowlog_internal.h#L5

Comment (stoeckmann) [reply lib/run_part.c]: I'd accept arguing about keeping perror after the fork, but simply writing into stderr from a fork isn't clean either. Otherwise the rational is that the logging infrastructure is there, both cases were available in that file and the majority in lib/ uses the logging infrastructure (as expected).

Comment (alejandro-colomar) [reply lib/run_part.c]: I don't say it's wrong, but the commit message should say something.  Something like that would work.

Comment (stoeckmann) [reply lib/run_part.c]: Adjusted

File: lib/run_part.c
Comment (alejandro-colomar) [inline lib/run_part.c:34]: While changing the perror after fork, it might be good to change the string to "fork: %s\n", just like the others, for consistency.

File: lib/run_part.c
Comment (alejandro-colomar) [inline lib/run_part.c]: I would rather do the opposite white-space change.  The tendency is to get rid of it.

**Review by alejandro-colomar — APPROVED** [review]
```
Reviewed-by: Alejandro Colomar <alx@kernel.org>
```

I'll let someone else merge.  Thanks!

File: lib/run_part.c
Comment (ikerexxe) [inline lib/run_part.c]: Can you add `failed` at the end to make it more understandable when reading the logs? Something like `fork failed: %s...`

File: lib/run_part.c
Comment (ikerexxe) [inline lib/run_part.c]: Same here, if not I fear it'll be difficult to understand the problem. Example `asprintf failed...`

**Review by ikerexxe — COMMENTED** [review]
Two minor comments. Once those are fixed I'll proceed to merge.

File: lib/run_part.c
Comment (alejandro-colomar) [reply lib/run_part.c]: The format used here is common in many programs:

```sh
$ cat /nonexistent
cat: /nonexistent: No such file or directory
```

The errno description already implies an error or failure.

File: lib/run_part.c
Comment (alejandro-colomar) [reply lib/run_part.c]:
```
asprintf: Cannot allocate memory
```

Should imply a failure.  :)

File: lib/run_part.c
Comment (stoeckmann) [reply lib/run_part.c]: I agree with @alejandro-colomar here. It's pretty much the same output you would get with the `perror` calls as they are in this file right now as well.

File: lib/run_part.c
Comment (ikerexxe) [reply lib/run_part.c]: You are right here

**Review by ikerexxe — APPROVED** [review]
LGTM! Thank you for the patches

**Pull request merged**
